core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/Changeset.java

```diff
@@ -176,12 +176,20 @@ boolean hasApproved(Resource subj, IRI pred, Value obj, Resource[] contexts) {
 
 	boolean hasDeprecated(Resource subj, IRI pred, Value obj, Resource[] contexts) {
 		assert !closed;
-		if (deprecated == null || deprecatedEmpty) {
+		if ((deprecated == null || deprecatedEmpty) && deprecatedContexts == null) {
 			return false;
 		}
 
 		boolean readLock = readWriteLock.readLock();
 		try {
+			if (deprecatedContexts != null) {
+				for (Resource context : contexts) {
+					if (deprecatedContexts.contains(context)) {
+						return true;
+					}
+				}
+			}
+
 			return deprecated.contains(subj, pred, obj, contexts);
```

Comment:
There was actually a bug in the Changeset code where it didn't handle deprecating a context correctly. I think this was something that I introduced in 4.1.0 with my Changeset optimizations.